test/stream.js

```diff
@@ -173,6 +174,11 @@ test('buffer: false > promise rejects when process returns non-zero', async t =>
 	t.is(exitCode, 2);
 });
 
+test('buffer: false > emits end event when promise is rejected', async t => {
+	const subprocess = execa('wrong command', {buffer: false, reject: false});
+	await t.notThrowsAsync(Promise.all([subprocess, pEvent(subprocess.stdout, 'end')]));
```

Comment:
Before the code fix above, the `end` event would never be emitted on `subprocess.stdout` in that test case.